package.json

```
@@ -174,6 +174,14 @@
       }
     ]
   },
+  "eslintIgnore": [
+    "./*.js",
+    "./*.d.ts",
+    "app",
+    "**/tests",
+    "**/__tests__",
+    "config"
```

```suggestion
    "./*.d.ts",
    "**/tests",
    "**/__tests__",
```

Ideally, I would like ESLint to report in most of these directories.

---

Since we want to run the lint command on `app`, `src` and `config`, but at the same time want `tsc` to compile files in `src` only, I introduced a separate `tsconfig.eslint.json` file for `eslint`, extending from the base `tsconfig.json` file. Do you think this makes good sense?

---

I'll have to check. I generally don't run the lint command, just let VSC warn me and I fix the warnings immediately.

---

hmm, how do you think about adding lint-staged and husky to lint before committing?

---

Considering the low amount of external PRs, I never really thought this would be necessary. We tried adding `husky` once, but it had issues and got reverted. (#2156, https://github.com/nextauthjs/next-auth/pull/2130#issuecomment-858145520)

Since it's mainly me who creates code, it seemed unnecessary until this point. If we can do it in a reliable way, I guess it would be alright.

---

Got it. After checking, please let me know if there's anything that should be changed in this PR.